exercises/book-store/tests/book-store.rs

```diff
@@ -0,0 +1,135 @@
+//! Tests for book-store
+//!
+//! Generated by [script][script] using [canonical data][canonical-data]
+//!
+//! [script]: https://github.com/exercism/rust/blob/master/bin/init_exercise.py
+//! [canonical-data]: https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/book-store/canonical_data.json
+
+extern crate book_store;
+use book_store::*;
+
+/// Process a single test case for the property `total`
+///
+/// All cases for the `total` property are implemented
+/// in terms of this function.
+///
+/// Note that you'll need to both name the expected transform which
+/// the student needs to write, and name the types of the inputs and outputs.
+/// While rustc _may_ be able to handle things properly given a working example,
+/// students will face confusing errors if the `I` and `O` types are not concrete.
+///
+/// Expected input format: ('basket', 'targetgrouping')
+fn process_total_case(input: (Vec<usize>, Vec<Vec<usize>>), expected: f64) {
```

Comment:
I note that `input.1` is never used, but I'm not particularly concerned. Might make one wonder "why not just *not* pass it into this function, then?"

Reply:
As commented above, the `input` tuple contains all the case properties which we don't know. In this case, I believe that the `targetgrouping` field was meant to be an _output_ hint.

I considered having the student return their groups and comparing them to the target groupings, but I decided against it. Checking the computed lowest price only gives the same assurance as checking it with the groups, but with a considerably simpler interface.